GitHubTeams.ps1

```diff
@@ -276,6 +314,407 @@ filter Get-GitHubTeamMember
+
+    .PARAMETER NoStatus
+        If this switch is specified, long-running commands will run on the main thread
+        with no commandline status update.  When not specified, those commands run in
+        the background, enabling the command prompt to provide status information.
+        If not supplied here, the DefaultNoStatus configuration property value will be used.
+
+    .OUTPUTS
+        PSCustomObject
+
+    .EXAMPLE
+        New-GitHubTeam -OrganizationName PowerShell -TeamName 'Developers'
+
+        Creates a new GitHub team called 'Developers' in the 'PowerShell' organization.
+#>
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSShouldProcess', '',
+        Justification = 'Methods called within here make use of PSShouldProcess, and the switch
+        is passed on to them inherently.')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSReviewUnusedParameter', '',
+        Justification = 'One or more parameters (like NoStatus) are only referenced by helper
+        methods which get access to it from the stack via Get-Variable -Scope 1.')]
+    [CmdletBinding(
+        SupportsShouldProcess,
+        PositionalBinding = $false
+    )]
+    param
+    (
+        [Parameter(Mandatory, Position = 1)]
+        [ValidateNotNullOrEmpty()]
+        [string] $OrganizationName,
+
+        [Parameter(Mandatory, Position = 2)]
+        [ValidateNotNullOrEmpty()]
+        [string] $TeamName,
+
+        [string] $Description,
+
+        [string[]] $Maintainers,
+
+        [string[]] $RepositoryName,
+
+        [ValidateSet('Secret','Closed')]
+        [string] $Privacy,
+
+        [string] $ParentTeamName,
+
+        [string] $AccessToken,
+
+        [switch] $NoStatus
+    )
+
+    Write-InvocationLog
+
+    $telemetryProperties = @{
+        OrganizationName = (Get-PiiSafeString -PlainText $OrganizationName)
+        TeamName = (Get-PiiSafeString -PlainText $TeamName)
+    }
+
+    $uriFragment = "/orgs/$OrganizationName/teams"
+
+    $hashBody = @{
+        name = $TeamName
+    }
+
+    if ($PSBoundParameters.ContainsKey('Description')) { $hashBody['description'] = $Description }
+    if ($PSBoundParameters.ContainsKey('Maintainers')) { $hashBody['maintainers'] = $Maintainers }
+    if ($PSBoundParameters.ContainsKey('RepositoryName')) { $hashBody['repo_names'] = $RepositoryName }
+    if ($PSBoundParameters.ContainsKey('Privacy')) { $hashBody['privacy'] = $Privacy.ToLower() }
+    if ($PSBoundParameters.ContainsKey('ParentTeamName'))
+    {
+        $getGitHubTeamParms = @{
+            OrganizationName = $OrganizationName
+            TeamName = $ParentTeamName
+            Whatif = $false
+            Confirm = $false
+        }
+        if ($PSBoundParameters.ContainsKey('AccessToken'))
+        {
+            $getGitHubTeamParms['AccessToken'] = $AccessToken
+        }
+        if ($PSBoundParameters.ContainsKey('NoStatus'))
+        {
+            $getGitHubTeamParms['NoStatus'] = $NoStatus
```

Comment:
This won't work right (I know because I made the same mistake in my Releases PR).  You'll see the bug during Pester test runs when `DefaultNoStatus` is set to `$true`.  You need to use `Resolve-ParameterWithDefaultConfigurationValue -Name NoStatus -ConfigValueName DefaultNoStatus`

Reply:
You're totally right.  My mistake.  With my issue, I was [directly referring to `$NoStatus`](https://github.com/microsoft/PowerShellForGitHub/pull/177/commits/03ec7d969a7b17b5a3de1373f3117627124df464#diff-b2345e64c8f3c38d6742b77af79129a0L951) which caused the issue.  You're guarding yourself here with the `ContainsKey` check.  So, you won't run afoul of the same issue.  You _could_ choose to use `Resolve-ParameterWithDefaultConfigurationValue` to avoid the check still, but keeping the check there still maintains the same end result.